code/game/gamemodes/miniantags/witchcraft/witch_procs.dm

```diff
@@ -0,0 +1,39 @@
+//Helper procs for witches.
+
+//To check if a mind is a witch. Standard syntax should be "is_witch([mind])".
+//Example: is_witch(H) to check if H, a mind, is a witch.
+/proc/is_witch(var/datum/mind/M)
+	if(!M || !istype(M) || !M.witch)
+		return 0
+	return 1
+
+//To check the elemental affinity of a mind. Standard syntax should be "get_affinity([mind], [define of affinity - check witch_datum.dm])".
+//Example: get_affinity(X, AFFINITY_WATER) to check if X, a mind, is attuned to water.
+/proc/check_affinity(var/datum/mind/M, var/req_affinity)
+	if(!M || !istype(M) || !M.witch)
+		return 0
+	var/datum/witch/W = M.witch
+	if(W.affinity == req_affinity)
```

Comment:
Just change this line to `if(istype(W.affinity, req_affinity))` and make the proc take a type path as argument. Change the defines to the type paths of the datums if you want.